### ISSUE TITLE
[Fabric] 1.21 JVM Crash? Game freezes and completely crashes

### ISSUE THREAD

**Issue by NovaLynxie** [IrisShaders/Iris#2445]
### Minecraft Version

Minecraft 1.21

### Iris Version

iris-1.7.3+mc1.21

### Sodium Version

sodium-0.5.11+mc1.21

### Operating System

Windows 11

### What is your GPU?

AMD Barcelo - Internal GPU [Lenovo] Radeon Vega

### Java Version

Java 21

### Reproduction Steps

1. Install Iris for Fabric 1.21
2. Load the game and enable any shaders you want to use.
3. Load into a world or a multiplayer server.
4. Play for a while; the game will eventually lock up, freeze for up to a minute, and finally crash.
5. Check the game console. It should return the following (or similar): `Process crashed with exitcode -1073740791.`

### Crash Report file and latest.log

```shell
No crash logs are available! See additional context for more details.
```

### Additional context

I have tried checking for driver updates for my GPU, but this has not worked.
At first, I thought this was a problem caused by the Voice Chat mod by Henkelmax, but later, I realized it might be a graphics issue, as I was misled by errors thrown by the computer, suggesting otherwise.
Latest game log: https://gist.github.com/NovaLynxie/284fb17a880f084fa1924bd133d612c3
hs_err_pidXXXXX.log: https://gist.github.com/NovaLynxie/a71d7361d7b921eef5d6dfb9fef8d44d
It is strange cause I was able to run shaders on their lowest settings with stable enough results, so I suspect Iris or the shader I tried to load is causing my problems.
Used the following shaders (any shader, it seems, eventually causes this):
BSL v8.2.09
Cyanide Shaders v1.1
Simplicissimus v1.3a
Simply Upscaled v0.1.8

**Comment by JustPixeled** [comment]
I had this same issue when rendering one part of the of Minescape server when shaders are enabled. Do you or the server use a resource pack with custom block models? I thought my crash was to do with them

**Comment by mcxiangdon** [comment]
I had the same issue too after reloading a resource pack in a world. Loaded ETF and EMF too.

**Comment by MeowLorrrd** [comment]
I've had this error too. A workaround I found is using the '[ForceGL20 mod](https://modrinth.com/mod/forcegl2.0-remapped)', which 'forces' the game to use OpenGL 2.0 (for me it actually sets it to version 4.6). It might decrease performance, but I haven't encountered this error since.

**Comment by IMS212** [comment]
This has been properly fixed in Iris 1.8 Beta 3.

**Comment by Mohorowe** [comment]
I had the same issue, causing a memory leak, filling my memory up to 100 leading to freeze and crash
